src/v/kafka/server/handlers/configs/config_response_utils.cc

```diff
@@ -0,0 +1,976 @@
+      resource,
+      result,
+      "log.segment.bytes",
+      config::shard_local_cfg().log_segment_size,
+      include_synonyms,
+      maybe_make_documentation(
+        include_documentation,
+        config::shard_local_cfg().log_segment_size.desc()),
+      &describe_as_string<size_t>);
+
+    add_broker_config_if_requested(
+      resource,
+      result,
+      "log.retention.bytes",
+      config::shard_local_cfg().retention_bytes,
+      include_synonyms,
+      maybe_make_documentation(
+        include_documentation,
+        config::shard_local_cfg().retention_bytes.desc()),
+      [](std::optional<size_t> sz) {
+          return ssx::sformat("{}", sz ? sz.value() : -1);
+      });
+
+    add_broker_config_if_requested(
+      resource,
+      result,
+      "log.retention.ms",
+      config::shard_local_cfg().log_retention_ms,
+      include_synonyms,
+      maybe_make_documentation(
+        include_documentation,
+        config::shard_local_cfg().log_retention_ms.desc()),
+      [](const std::optional<std::chrono::milliseconds>& ret) {
+          return ssx::sformat("{}", ret.value_or(-1ms).count());
+      });
+
+    add_broker_config_if_requested(
+      resource,
+      result,
+      "num.partitions",
+      config::shard_local_cfg().default_topic_partitions,
+      include_synonyms,
+      maybe_make_documentation(
+        include_documentation,
+        config::shard_local_cfg().default_topic_partitions.desc()),
+      &describe_as_string<int32_t>);
+
+    add_broker_config_if_requested(
+      resource,
+      result,
+      "default.replication.factor",
+      config::shard_local_cfg().default_topic_replication,
+      include_synonyms,
+      maybe_make_documentation(
+        include_documentation,
+        config::shard_local_cfg().default_topic_replication.desc()),
+      &describe_as_string<int16_t>);
+
+    add_broker_config_if_requested(
+      resource,
+      result,
+      "log.dirs",
+      config::node().data_directory,
+      include_synonyms,
+      maybe_make_documentation(
+        include_documentation, config::node().data_directory.desc()),
+      [](const config::data_directory_path& path) {
+          return path.as_sstring();
+      });
+
+    add_broker_config_if_requested(
+      resource,
+      result,
+      "auto.create.topics.enable",
+      config::shard_local_cfg().auto_create_topics_enabled,
+      include_synonyms,
+      maybe_make_documentation(
+        include_documentation,
+        config::shard_local_cfg().auto_create_topics_enabled.desc()),
+      &describe_as_string<bool>);
+}
+
+int64_t describe_retention_duration(
+  tristate<std::chrono::milliseconds>& overrides,
+  std::optional<std::chrono::milliseconds> def) {
+    if (overrides.is_disabled()) {
+        return -1;
+    }
+    if (overrides.has_optional_value()) {
+        return overrides.value().count();
+    }
+
+    return def ? def->count() : -1;
+}
+int64_t describe_retention_bytes(
```

Comment:
is this used anywhere?

Reply:
Removed: https://github.com/redpanda-data/redpanda/pull/16922/commits/a56a5367ee60ebe6e71b679980bf82902da4b24e